main/preload.ts

```diff
@@ -47,6 +49,25 @@ const ipc = {
     });
   },
 
+  on(channel: string, callback: (...args: unknown[]) => void) {
+    const subscription = (_event: IpcRendererEvent, ...args: unknown[]) =>
+      callback(...args);
+    ipcRenderer.on(channel, subscription);
+
+    return () => {
+      ipcRenderer.removeListener(channel, subscription);
+    };
+  },
+
+  removeAllListeners(channel: string) {
+    ipcRenderer.removeAllListeners(channel);
+  },
+
+  async downloadUpdate(): Promise<unknown> {
+    const result: unknown = await ipcRenderer.invoke('download-update');
```

Comment:
Please set the value of `download-update`  in the `IPC_ACTIONS` enum and use it here. 

You can see other references of its usage if you could lookup this keyword `IPC_ACTIONS`